src/index.d.ts

```diff
@@ -0,0 +1,108 @@
+type ValueOf<T> = T[keyof T];
+
+type CHILD_SEED_TYPES = {
+  OWNERSHIP: 'ownership',
+  TRANSFER: 'transfer',
+  SPAWN: 'spawn',
+  VOTING: 'voting',
+  MANAGEMENT: 'management',
+  NETWORK: 'network',
+  BITCOIN_MAINNET: 'bitcoinMainnet',
+  BITCOIN_TESTNET: 'bitcoinTestnet',
+}
+
+type ChildSeedType = ValueOf<CHILD_SEED_TYPES>;
+
+interface NetworkKeys {
+  crypt: {
+    private: string;
+    public: string;
+  };
+  auth: {
+    private: string;
+    public: string;
+  };
+}
+
+interface WalletNodeKeys {
+  public: string;
+  private: string;
+  chain: string;
+  address: string;
+}
+
+interface WalletNode {
+  type: string;
+  seed: string;
+  keys: WalletNodeKeys;
+  derivationPath: string;
+}
+
+interface BitcoinWallet extends WalletNode { }
+
+interface WalletConfig {
+  ticket: string;
+  ship: number;
+  passphrase?: string;
+  boot?: boolean;
+}
+
+interface UrbitWallet {
+  meta: {
+    generator: {
+      name: string;
+      version: string;
+    };
+    spec: string;
+    ship: string;
+    patp: string;
+    tier: string;
+    passphrase: string;
+  };
+  masterSeed: string;
+  ownership: WalletNode;
+  management: WalletNode;
+  transfer: WalletNode;
+  network:
+  | {
+    type: string;
+    seed: string;
+    keys: string;
+  }
+  | {};
```

Comment:
Why not `null` for the second case here?

Reply:
`#generateWallet` accepts an optional config argument, `boot`:
https://github.com/urbit/urbit-key-generation/blob/master/src/index.js#L612

which is consumed here:
https://github.com/urbit/urbit-key-generation/blob/master/src/index.js#L676

if it's `false`, an empty object is returned for `network`

Reply:
Arguably we should return null instead of the empty object, but I guess that's a breaking api change. Definitions here all seem fine!